Attempt at fixing pdass intermittent failure on CIVET

recuero opened the pull request:
## Reason
Avoid intermittent failure of a test

## Design
No changes. Reduce mesh size, change solver settings.

## Impact
More robust tests, less hassle for developers, who need to invalidate the test to get a clean sheet.


Refs. #17495

recuero committed:
Attempt at fixing pdass intermittent failure on CIVET Refs. #17495

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/754111/) on 7c6b67b wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/18013/site)

This comment will be updated on new commits.

arovinelli commented:
@recuero I would suggest adding `max_parallel=min_parallel` in the failing test, leave the same tolerance and regold. I suspect this behavior is caused by different partitioning, which depends on the number of tasks (CPU).

recuero committed:
Address Andrea's review.

Enforce same number of processors to avoid different results due to variability in domain decomposition.

lindsayad commented:
> @recuero I would suggest adding `max_parallel=min_parallel` in the failing test, leave the same tolerance and regold. I suspect this behavior is caused by different partitioning, which depends on the number of tasks (CPU).

I disagree with this. Ideally our tests should be robust enough that the solution is stable across different processor counts. Moreover, this test itself has been shown to pass at various times with `np = 4` and fail at various times with `np = 4` so just keeping the processor count enforced at 4 is not guaranteed to always elicit a pass. I think making the problem a little coarser like @recuero did, which will make the solve a bit easier, and possibly adjusting solver settings are the right things to do.

Although I'm not sure whether adjusting `l_tol` is the right way to go. It *might* be. For example if a tighter linear solve helps ensure that you don't have a bad Newton update, then that's definitely a good thing. However, sometimes when I run with `-ksp_monitor_true_residual` with PJFNK, a bad Newton update often corresponds to a linear solve in which the *true* residual actually *increased* over the course of the linear iterations. I might investigate that here. If the *true* residual is always going down, then decreasing `l_tol` is a good thing.

recuero committed:
Address Alex's comment.

Decreasing `l_tol` does not guarantee better convergence. Especially for this test, where the true residual goes up during
the first nonliner iteration. Given that the first nonliner iteration is responsible for the sporadic cuts in the time
step, I am reverting here the modification of this value. I am keeping the increase in nonlinear iterations, as in those
instances where the test fails, it seems right on track to convergence in a few more nonlinear iterations.

recuero commented:
Thanks @lindsayad. For this test, tightening `l_tol` makes the true residual go down for all nonlinear iterations that aren't the first one. For the first one, the true residual does go up a bit over the last couple of linear iterations, which doesn't seem to be very significant. 

For other tests with contact, I have seen that tightening `l_tol` improves the overall convergence behavior. That said, after analyzing `-ksp_monitor_true_residual` for this problem, I think that doing that is counterproductive, as the first nonlinear iteration appears to be responsible for the ultimate cut in time step.
